Sources/eppo/UniversalFlagConfig.swift

```diff
@@ -0,0 +1,145 @@
+            case .valueNotFound(_, let context):
+                throw UniversalFlagConfigError.parsingError("Value not found: \(context.debugDescription)")
+            case .keyNotFound(let key, let context):
+                throw UniversalFlagConfigError.parsingError("Key not found: \(key.stringValue) - \(context.debugDescription)")
+            case .dataCorrupted(let context):
+                throw UniversalFlagConfigError.parsingError("Data corrupted: \(context.debugDescription)")
+            default:
+                throw UniversalFlagConfigError.parsingError("JSON parsing error: \(error.localizedDescription)")
+            }
+        } catch {
+            throw UniversalFlagConfigError.parsingError("Unexpected error: \(error.localizedDescription)")
+        }
+    }
+}
+
+// enums
+enum UniversalFlagConfigError: Error, CustomNSError, LocalizedError {
+    case notUTF8Encoded(String)
+    case parsingError(String)
+
+    static var errorDomain: String { return "UniversalFlagConfigError" }
+    
+    var errorCode: Int {
+        switch self {
+        case .notUTF8Encoded:
+            return 100 
+        case .parsingError:
+            return 101
+        }
+    }
+
+    var errorDescription: String? {
+        switch self {
+        case .notUTF8Encoded(let message):
+            return message
+        case .parsingError(let message):
+            return message
+        }
+    }
+}
+
+enum UFC_VariationType: String, Decodable {
+      case boolean = "BOOLEAN"
+      case integer = "INTEGER"
+      case json = "JSON"
+      case numeric = "NUMERIC"
+      case string = "STRING"    
+}
+
+enum UFC_AlgorithmType: String, Decodable {
+    case constant = "CONSTANT"
+    case contextualBandit = "CONTEXTUAL_BANDIT"
+}
+
+enum UFC_RuleConditionOperator: String, Decodable {
+  case lessThan = "LT"
+  case lessThanEqual = "LTE"
+  case greaterThan = "GT"
+  case greaterThanEqual = "GTE"
+  case matches = "MATCHES"
+  case oneOf = "ONE_OF"
+  case notOneOf = "NOT_ONE_OF"
+  case isNull = "IS_NULL"
+}
+
+// models
+
+public struct UFC_Flag : Decodable {
+    let key: String;
+    let enabled: Bool;
+    let variationType: UFC_VariationType;
+    let variations: [String: UFC_Variation];
+    let allocations: [UFC_Allocation];
+    let totalShards: Int;
+}
+
+public struct UFC_Variation : Decodable  {
+    let key: String;
+    let value: EppoValue;
+    let algorithmType: UFC_AlgorithmType?
+}
+
+public struct UFC_Allocation : Decodable {
+    let key: String;
+    let rules: [UFC_Rule]?;
+    let startAt: Date?;
+    let endAt: Date?;
+    let splits: [UFC_Split];
+    let doLog: Bool;
+}
+
+public struct UFC_Rule : Decodable {
+    let conditions: [UFC_TargetingRuleCondition];
+}
+
+public struct UFC_TargetingRuleCondition : Decodable {
+    let `operator`: UFC_RuleConditionOperator;
+    let attribute: String;
+    let value: EppoValue;
```

Comment:
What is EppoValue for you? In other SDKs it refers to the values that the SDK can return and does not include Array[String]

Reply:
It handles all types: numeric, string and array of strings; originally created for the variation value and typed values. 

I noticed in the python implementation that this was split into two classes to more directly match the API UFC response so they were more limited but as soon as you are returning multiple types in a json value it's not possible to know what to expect so I am accepting all.